src/modules/simulation/gz_bridge/GZMixingInterfaceServo.cpp

```diff
@@ -33,6 +33,124 @@
 
 #include "GZMixingInterfaceServo.hpp"
 
+float
+GZMixingInterfaceServo::get_tl_angle_max(const size_t index)
+{
+	switch (index) {
+	case 0: return _ca_sv_tl_max_a_0.get();
+
+	case 1: return _ca_sv_tl_max_a_1.get();
+
+	case 2: return _ca_sv_tl_max_a_2.get();
+
+	case 3: return _ca_sv_tl_max_a_3.get();
+
+	default: return 0.f;
+	}
+}
+
+float
+GZMixingInterfaceServo::get_tl_angle_min(const size_t index)
+{
+	switch (index) {
+	case 0: return _ca_sv_tl_min_a_0.get();
+
+	case 1: return _ca_sv_tl_min_a_1.get();
+
+	case 2: return _ca_sv_tl_min_a_2.get();
+
+	case 3: return _ca_sv_tl_min_a_3.get();
+
+	default: return 0.f;
+	}
+}
+
+float
+GZMixingInterfaceServo::get_cs_angle_max(const size_t index)
+{
+	switch (index) {
+	case 0: return _ca_sv_cs_max_a_0.get();
+
+	case 1: return _ca_sv_cs_max_a_1.get();
+
+	case 2: return _ca_sv_cs_max_a_2.get();
+
+	case 3: return _ca_sv_cs_max_a_3.get();
+
+	case 4: return _ca_sv_cs_max_a_4.get();
+
+	case 5: return _ca_sv_cs_max_a_5.get();
+
+	case 6: return _ca_sv_cs_max_a_6.get();
+
+	case 7: return _ca_sv_cs_max_a_7.get();
+
+	default: return 0.f;
+	}
+}
+
+float
+GZMixingInterfaceServo::get_cs_angle_min(const size_t index)
+{
+	switch (index) {
+	case 0: return _ca_sv_cs_min_a_0.get();
+
+	case 1: return _ca_sv_cs_min_a_1.get();
+
+	case 2: return _ca_sv_cs_min_a_2.get();
+
+	case 3: return _ca_sv_cs_min_a_3.get();
+
+	case 4: return _ca_sv_cs_min_a_4.get();
+
+	case 5: return _ca_sv_cs_min_a_5.get();
+
+	case 6: return _ca_sv_cs_min_a_6.get();
+
+	case 7: return _ca_sv_cs_min_a_7.get();
+
+	default: return 0.f;
+	}
+}
+
+int GZMixingInterfaceServo::get_active_output_count()
+{
+	return _control_surface_count.get() + _tilt_count.get();
+}
+
+bool GZMixingInterfaceServo::is_tiltrotor(const int index)
+{
+	if (_tilt_count.get() < 1) {
+		return false;
+	}
+
+	return (index > _control_surface_count.get() - 1) && (index < get_active_output_count());
+}
+
+double GZMixingInterfaceServo::get_angle_max_wrapper(const int index)
+{
+	if (index >= get_active_output_count()) {
+		return 0.;
```

Comment:
same here? angle zero would be an valid value, but in this context its used as out of range, NAN is generally used for that.

Reply:
Removed this code block